### PR TITLE
Remove .exe suffix when cross-compiling on Windows

### PR DESCRIPTION
**Pull request by sryze**
When compiling GItea for Linux on Windows, you get a `gitea.exe` file as output, but because it's a Linux executable, the `.exe` extension is unnecessary.

This PR adds a check for `GOOS` environment variable in addition to `OS`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,10 +49,15 @@ ifeq ($(HAS_GO), yes)
...
 
-ifeq ($(OS), Windows_NT)
-	GOFLAGS := -v -buildmode=exe
-	EXECUTABLE ?= gitea.exe
-else ifeq ($(OS), Windows)
+ifeq (Windows, $(findstring Windows,$(OS)))
+	ifeq (,$(GOOS))
+		IS_WINDOWS := yes
+	endif
+	ifeq (windows,$(GOOS))
```
Comment (silverwind) [inline Makefile]:
```suggestion
	ifeq ($(GOOS),)
		IS_WINDOWS := yes
	endif
	ifeq ($(GOOS),windows)
```

No yoda conditions.

Comment (sryze) [reply Makefile]: Thanks, updated

(didn't know it was called Yoda conditions 😄 )

Comment (silverwind) [reply Makefile]: https://en.wikipedia.org/wiki/Yoda_conditions 😆

**Review by silverwind — APPROVED** [review]

**Comment by sryze** [comment]
@lng2020 Thanks, I haven't thought of the reverse situation for some reason. Updated.

**Review by nanguanlin6 — APPROVED** [review]
BTW, Is there a certain reason to replace `findstring Windows`  with `Windows_NT`? I did the search and those two ways seem the same.

**Comment by silverwind** [comment]
> BTW, Is there a certain reason to replace `findstring Windows` with `Windows_NT`? I did the search and those two ways seem the same.

I guess the previous exact match is still better, even if the name is not accurate. I was unable to find any docs on this `OS` variable anywhere, I assume it is provided by Make itself?

**Comment by sryze** [comment]
It is defined by default on Windows by the system. I couldn't find any docs for this on MSDN, but it's mentioned in a Wikipedia article [here](https://en.wikipedia.org/wiki/Environment_variable)

> %OS%
    The %OS% variable contains a symbolic name of the operating system family to distinguish between differing feature sets in [batchjobs](https://en.wikipedia.org/wiki/Batchjob). It resembles an identically named environment variable [%OS%](https://en.wikipedia.org/wiki/Environment_variable#%OS%) found in all DOS-related operating systems of [Digital Research](https://en.wikipedia.org/wiki/Digital_Research)-origin like Concurrent DOS, [Multiuser DOS](https://en.wikipedia.org/wiki/Multiuser_DOS), REAL/32, [DOS Plus](https://en.wikipedia.org/wiki/DOS_Plus), [DR DOS](https://en.wikipedia.org/wiki/DR_DOS), Novell DOS and OpenDOS. %OS% always holds the string "Windows_NT" on the [Windows NT family](https://en.wikipedia.org/wiki/Windows_NT_family).[[26]](https://en.wikipedia.org/wiki/Environment_variable#cite_note-Ten_XP-28)

It seems that it should be safe to just check for equality with `Windows_NT`, I'm not sure why the Makefile had an additional `Windows` case. Do you think it should be a `ifeq $(OS),Windows_NT` only?

**Comment by sryze** [comment]
Looks like it returns `Windows` on some machines, accoding to this issue: https://github.com/go-gitea/gitea/pull/16007

**Comment by silverwind** [comment]
> Looks like it returns `Windows` on some machines, accoding to this issue: #16007

In this case the `findstr` is good. If only Make had a way to test whether string "starts with", but it seems it doesn't.

File: Makefile [hunk 2]
```diff
@@ -49,10 +49,14 @@ ifeq ($(HAS_GO), yes)
 	CGO_CFLAGS ?= $(shell $(GO) env CGO_CFLAGS) $(CGO_EXTRA_CFLAGS)
 endif
 
-ifeq ($(OS), Windows_NT)
-	GOFLAGS := -v -buildmode=exe
-	EXECUTABLE ?= gitea.exe
-else ifeq ($(OS), Windows)
+ifeq ($(GOOS),windows)
+	IS_WINDOWS := yes
+else ifeq ($(findstring Windows,$(OS)),Windows)
```
Comment (silverwind) [inline Makefile]:
```suggestion
else ifeq ($(patsubst Windows%,,$(OS)),)
```

Test with this:

```make
OS := Windows_NT

all:
ifeq ($(patsubst Windows%,,$(OS)),)
	@echo "$(OS) starts with Windows"
else
	@echo "$(OS) doesn't start with Windows"
endif

Comment (sryze) [reply Makefile]: I agree, this is better. Updated.

**Review by silverwind — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (sryze) [reply Makefile]: Actually this doesn't work when `OS` is empty

Comment (silverwind) [reply Makefile]: A bit of a hack but this works with empty as well:

```make
ifeq ($(patsubst Windows%,x,$(OS)),x)

File: Makefile [hunk 3]
```diff
@@ -49,10 +49,16 @@ ifeq ($(HAS_GO), yes)
...
-ifeq ($(OS), Windows_NT)
-	GOFLAGS := -v -buildmode=exe
-	EXECUTABLE ?= gitea.exe
-else ifeq ($(OS), Windows)
+ifeq ($(GOOS),windows)
+	IS_WINDOWS := yes
+else ifeq ($(GOOS),)
+	ifeq ($(OS),Windows_NT)
+		IS_WINDOWS := yes
+	else ifeq ($(OS),Windows)
```
Comment (silverwind) [inline Makefile:57]:
```suggestion
	ifeq ($(patsubst Windows%,x,$(OS)),x) # if string is not empty and starts with "Windows"
```

File: Makefile (hunk 2 above)
Comment (silverwind) [reply Makefile]: This is clearer:

```make
ifeq ($(patsubst Windows%,Windows,$(OS)),Windows)

Comment (silverwind) [reply Makefile]: On the other hand it fails with spaces `OS = Windows NT`. Let's keep the current version.

**Pull request merged**